hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/InsertProcessorSupplier.java

```diff
@@ -130,4 +139,31 @@ private Document rowToDoc(JetSqlRow row) {
         return doc;
     }
 
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeString(connectionString);
+        out.writeString(databaseName);
+        out.writeString(collectionName);
+        out.writeStringArray(paths);
+        out.writeObject(writeMode);
+        out.writeObject(types);
+        var typesLocal = externalTypes == null ? new BsonType[0] : externalTypes;
```

Comment:
When externalTypes is null this will throw NPE

Reply:
I changed external type in MongoTableField so that is now mandatory.

Reply:
Can you also add some tests for serialization. A simple one to serialize a supplier and deserialize it. Just to make sure it works

Reply:
@orcunc The serialization happens during Mongo connector tests so we know it works, do you still suggest to create a separate test for that?

Reply:
If it is executed then no need